tests/integration/targets/lookup_hashi_vault/lookup_hashi_vault/tasks/main.yml

```diff
@@ -6,7 +6,7 @@
 
 - name: Install Hashi Vault on controlled node and test
   vars:
-    vault_version: '0.11.0'
```

Comment:
@briantist After banging my head aginst the wall, I suspect that the failing test might be because of a very old Vault version. I tried to set a more recent version, but the download is failing atm. WDYT? Is there a chance that you can assist me in trying a newer version of Vault? Any particular reason why the tests are using an over two years old version?

Reply:
Ah, yes I've recently noticed the ancient vault version too, see discussion in #1018 .

The download fails because the binary was uploaded to an S3 bucket related to the Ansible project, but no newer ones were put there. If you suspect that's the problem, you can temporarily replace the whole URL and download directly from Hashicorp: https://releases.hashicorp.com/vault/

That should unblock you, but I imagine that before this gets merged, someone will have to upload a modern binary to the S3 bucket so that it can be referenced there.

Reply:
@briantist you probably have to provide a specific list of files and a specific list of names they should have on S3 to someone who has access (like @gundalow) to increase the chance that something happens.

Reply:
I had an idea of providing a PR to fix https://github.com/ansible-collections/community.general/issues/1018. 😄 Would it be a problem to change the download links to download directly from the source? I.e. https://releases.hashicorp.com/vault/... On the Vault download page, it states "Bandwidth courtesy of fastly".

Reply:
I guess we can start with that, and before merging the PR making sure the files are uploaded and used from S3.

Reply:
Erik the discussion in 1018 covers that somewhat. Typically in Ansible using external links in the CI is frowned upon, because they are outside the project's control and if something changes with them (downtime, path changes, etc.) they cause all CI runs to fail. As Felix pointed out, S3 can do down too, but at least for the time being, all of the CI runs in AWS so it's at least adjacent, and we (well, "they" as in RedHat/Ansible) control the paths in the bucket.

So as Felix just responded, we need to put newer versions in our bucket for now. Part of what I raised in 1018 was the idea of testing against multiple versions, as I'm not sure it's best to always go newest or only test some specific older version, but we do have to keep load on the test runner in mind.

To balance, I'd like to do 2 or 3 versions if possible, but anyway we should continue that discussion in #1018 .

Reply:
No luck with the latest version of Vault (1.6.0). So for the moment I am drawing blanks. There must be something wrong in the code or test! If you have any suggestions, please let me know.... 😢

Reply:
`Failure expected when unauthorized secret is read` is the test that's "failing" (in this case, succeeding when failure was expected), while the other 2 expected failures work as intended. Is it maybe an issue of the policy applied to the secret that makes the JWT authorized when we thought it wasn't?

Reply:
I am officially giving up fixing the failing test, and atm it is commented out (with a FIXME/TODO) - but I have problem removing it completely. And the more I have been struggeling with it, I believe that some of these negative tests do not belong in this lookup plugin. If Vault is allowing access to secrets with an explicit deny-policy, I think that must be a bug in Vault - and should of course be tested (and fixed) over there. But I can't really imagine they missed out a such important test - that is why I have tried my best to figure out problem in context of this PR. But no luck....